onboarding/source/amb_light_service.c

```diff
@@ -1,39 +1,125 @@
 #include "amb_light_service.h"
 #include "serial_io.h"
+#include <stdio.h>
 
 #include <adc.h>
 #include <sci.h>
+#include <FreeRTOS.h>
+#include <os_task.h>
+
+#include <os_queue.h>
+#include <os_projdefs.h>
+#include <string.h>
+
+#define Light_Sensor 	6U
 
 /* USER CODE BEGIN */
 // Include any additional headers and global variables here
 
+static TaskHandle_t lightServiceHandle = NULL;
+static QueueHandle_t queue = NULL;
 /* USER CODE END */
 
 /**
  * @brief Task that waits for MEASURE_LIGHT event and then prints the ambient light value to the serial port.
  * @param pvParameters Task parameters
  */
+
+void adcStartConversion_selChn(adcBASE_t *adc, unsigned channel, unsigned fifo_size, unsigned group);
+void adcGetSingleData(adcBASE_t *adc, unsigned group, adcData_t *data);
+uint32 Get_Light_Sensor_data(void);
 static void lightServiceTask(void * pvParameters);
+uint8_t sendToLightServiceQueue(light_event_t *event);
 
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    
+    if (queue == NULL) {
+        queue = xQueueCreate(1, sizeof(MEASURE_LIGHT));
+    }
 
-    /* USER CODE END */
+    if (lightServiceHandle == NULL) {
+        // Create controller task
+        xReturned = xTaskCreate(lightServiceTask,             /* Function that implements the task. */
+                                CONTROLLER_NAME_LIGHT,            /* Text name for the task. */
+                                CONTROLLER_STACK_SIZE,      /* Stack size in words, not bytes. */
+                                NULL,                       /* Parameter passed into the task. */
+                                CONTROLLER_PRIORITY,        /* Priority at which the task is created. */
+                                &lightServiceHandle);     /* Used to pass out the created task's handle. */
+    }
+
+    if(xReturned == pdFAIL){
+        unsigned char err[] = "Error Starting Light Service";
+        sciPrintText(sciREG, (unsigned char*) err, strlen((const char*) err));
```

Comment:
I'd suggest printing to `scilinREG` instead. This will make it easier to read the serial data on the RM46 LaunchPad because its built-in USB-serial functionality uses `scilinREG`. If you want to use `sciREG`, you'd need to connect a separate USB-UART converter to the board.